pkg/injector/test/test_helpers.go

```diff
@@ -0,0 +1,102 @@
+package test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+// All the YAML files listed above are in this sub-directory
+const directoryForYAMLFiles = "test_fixtures"
+
+var log = logger.New("sidecar-injector")
+
+// LoadExpectedEnvoyYAML loads the expectation for a given test from the file system. This must run within ginkgo.It()
+func LoadExpectedEnvoyYAML(filename string) string {
+	filePath := path.Join(directoryForYAMLFiles, filename)
+
+	// The filename will contain the name of the function by convention
+	log.Info().Msgf("Loading expected test output from file %s", filename)
+
+	if _, err := os.Stat(filepath.Clean(filePath)); os.IsNotExist(err) {
```

Comment:
if your aim is to create a new empty file then you can use os.Create directly which will truncate the file if exists.

Reply:
Removed the creation of a file. This was meant to be a helper - when a new test is written, but the expectation does not exist - it would be nice to create an empty file ready for the expectation YAML.
But this complicates the test helper.
The required file path is already logged - so the test writer can `touch filePath` it and add the expected yaml.